### ISSUE TITLE
Question about the input shape to TransformerLayer in RaygunDecoder

### ISSUE THREAD

**Issue by yuzimu1** [rohitsinghlab/raygun#8]
Hello,

I have a question regarding the input shape to the TransformerLayer inside esmdecoder. According to the documentation, the input shape for the TransformerLayer is expected to be (seq_len, batch, dim). However, in the Raygun model, the output from the decoder is:

`out = self.decoder(fixedembs, lengths)  # output shape: (batch, seq_len, dim)`

The shape of out is (batch, seq_len, dim) rather than the expected (seq_len, batch, dim).

Later, out is directly passed to` logits = self.esmdecoder(out)`, which involves the TransformerLayer.

Is this direct pass of out to esmdecoder without changing its shape ((batch, seq_len, dim)) going to cause any issues? Shouldn't we convert the shape of out to (seq_len, batch, dim) before passing it to esmdecoder?

Thank you for your time and help!

Best regards,

**Comment by kap-devkota** [comment]
Apologies. The input to the model should be of the form `(batch_size, seq_len, dim)` (i.e. batch_first). Additionally, if the batch size is greater than 1, an additional mask parameter of dims `(batch_size, seq_len)` should also be provided. 

I will update the documentation and fix this error. Thanks for pointing that out.
